src/alda/lisp/attributes.clj

```diff
@@ -74,3 +74,15 @@
   :aliases [:pan]
   :initial-val 0.5
   :transform percentage)
+
+(defn- parse-key-signature
+  "Transforms a key signature ((:c :sharp :sharp) (:d :flat)) into a letter
+   to accidentals map {:c (:sharp :sharp), :d (:flat)}."
+  [pitches]
+  (into {} (map (fn [pitch] {(first pitch) (rest pitch)}) pitches)))
+
+(defattribute key-signature
+   "The key in which the current instrument is playing."
+   :aliases [:key]
+   :initial-val {}
+   :transform (fn [val] (constantly (parse-key-signature (rest val)))))
```

Comment:
I have a feeling this problem might solve itself if you try hiding the `:pitch-notation` node like I mentioned. It would at least help us simplify things a bit, so we can figure out why exactly `:key-signature` is ending up in there. I'll bet it has something to do with the Instaparse transformation function for `:pitch-notation`